src/pages/chat/chat.tsx

```diff
@@ -0,0 +1,197 @@
+
+      if (lastMessageIndex >= 0) {
+        // 모든 메시지를 순회하며 showIcon을 설정
+        for (let i = 0; i < lastMessageIndex; i++) {
+          const currentSender = newMessages[i].sender;
+          const nextSender = newMessages[i + 1].sender;
+          newMessages[i].showIcon = currentSender !== nextSender;
+        }
+        // 마지막 메시지는 항상 showIcon을 true 로 설정
+        newMessages[lastMessageIndex].showIcon = true;
+      }
+
+      return newMessages;
+    });
+  }, [nowUser]);
+
+  if (!chatId) {
+    return <div>없는 친구 ID입니다.</div>;
+  }
+
+  const handleSendMessage = (content: string) => {
+    const newMessage: Message = {
+      id: messageId,
+      sender: nowUser.name,
+      content,
+      showIcon: false,
+      timestamp: new Date(), // 현재 시간 추가
+    };
+
+    // 메세지가 추가됐을 때 가장 마지막 index의 메세지와 비교하여 showIcon 업데이트
+    if (
+      messages.length > 0 &&
+      messages[messages.length - 1].sender !== nowUser.name
+    ) {
+      messages[messages.length - 1].showIcon = true;
+    }
+
+    const updatedMessages = [...messages, newMessage];
+    setMessages(updatedMessages);
+
+    // 다음 메시지를 위해 messageId 업데이트
+    setMessageId(messageId + 1);
+    // 로컬 스토리지에 채팅 데이터 저장
+    const savedChats = JSON.parse(
+      localStorage.getItem("chatMessages") || JSON.stringify(chatData)
+    ) as ChatData;
+
+    savedChats[chatId] = updatedMessages;
+    localStorage.setItem("chatMessages", JSON.stringify(savedChats));
+  };
+  const changeUser = () => {
+    setNowUser((prev) =>
+      prev.name === userData.users[0].name
+        ? userData.users[chatIdNumber - 1]
+        : userData.users[0]
+    );
+
+    setMessages((prevMessages) => {
+      const newMessages = [...prevMessages];
+      const lastMessageIndex = newMessages.length - 1;
+      //userChange 동시에 여러번 누르면 아이콘 동시에 뜨는것 방지하기 위해
+      if (lastMessageIndex >= 1) {
+        const lastMessageSender = newMessages[lastMessageIndex].sender;
+        const secondLastMessageSender =
+          newMessages[lastMessageIndex - 1].sender;
+
+        // 직전 메시지와 비교하여 showIcon을 설정합니다.
+        if (
+          secondLastMessageSender === lastMessageSender &&
+          newMessages[lastMessageIndex - 1].showIcon
+        ) {
+          newMessages[lastMessageIndex - 1].showIcon = false;
+        }
+        newMessages[lastMessageIndex].showIcon = true;
+      }
+
+      return newMessages;
+    });
+  };
+
+  //chatId를 integer 값으로 바꿔준 결과
+  const chatIdNumber = +chatId;
+  const partner =
+    nowUser.name === userData.users[0].name
+      ? userData.users[chatIdNumber - 1]
+      : userData.users[0];
+
+  const messageContainers = messages.map((message: Message, index) => {
+    const isCurrentUser = message.sender === nowUser.name;
+
+    return (
+      <Message
+        key={message.id}
+        sender={message.sender}
+        content={message.content}
+        nowUser={nowUser.name}
+        showIcon={message.showIcon && !isCurrentUser}
+      />
+    );
+  });
```

Comment:
showIcon을 판단할 때, 위처럼 따로 로직을 만들 수도 있지만, map 안에서 index를 통해 다음 message의 sender와 비교하여 sender가 바뀌는 시점에 showIcon을 true로 설정해주는 방법을 사용해도 좋을 것 같아요!
```suggestion
  const messageContainers = messages.map((message: Message, index) => {
    const isCurrentUser = message.sender === nowUser.name;

    return (
      <Message
        key={message.id}
        sender={message.sender}
        content={message.content}
        nowUser={nowUser.name}
        showIcon={message.sender !== messages[index + 1].sender && !isCurrentUser}
      />
    );
  });
```
이런 식으로요!